infra.yaml

```diff
@@ -265,7 +265,12 @@ resources:
   host:
     type: OS::Nova::Server
     properties:
-      name: {get_param: hostname}
+      name:
+        str_replace:
+          template: "HOST.DOMAIN"
+          params:
+            HOST: {get_param: hostname}
+            DOMAIN: {get_param: domain_name}
```

Comment:
What the Openstack cloud provider does is that it queries the Openstack API, looking for an instance with the hostname of the virtual machine.